### PR TITLE
pd: always update client

### PR DESCRIPTION
**Pull request by overvenus**
For one node PD cluster, tikv pd client will fail to make a request if the pd leader restarts. Similarly, if the member response doesn't change, tikv pd client will always skip updating the inner grpc client.

PTAL @siddontang @nolouch @BusyJay @zhangjinpeng1987

### CODE REVIEW

File: src/pd/async/util.rs [hunk 1]
```diff
@@ -106,14 +106,11 @@ impl<Req, Resp, F> Request<Req, Resp, F>
...
-        let members = self.inner.rl().members.clone();
-        match try_connect_leader(&members) {
+        match try_connect_leader(&self.inner.rl().members) {
             Ok((client, members)) => {
                 let mut inner = self.inner.wl();
-                if members != inner.members {
-                    inner.client = client;
-                    inner.members = members;
-                }
+                inner.client = client;
```
Comment (BusyJay) [inline src/pd/async/util.rs]: Any tests to cover this?

**Comment by siddontang** [comment]
any update? @overvenus

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @disksing

File: tests/pd/test_rpc_client.rs [hunk 2]
```diff
@@ -149,6 +149,57 @@ fn test_retry_async() {
...
+    drop(_server_b);
+    drop(_server_c);
+    // Restart them again.
+    let _server_a = MockServer::run("127.0.0.1:42978", se.clone(), Some(se.clone()));
+    let _server_b = MockServer::run("127.0.0.1:52978", se.clone(), Some(se.clone()));
+    let _server_c = MockServer::run("127.0.0.1:62978", se.clone(), Some(se.clone()));
+    thread::sleep(Duration::from_secs(1));
+
+    let region = client.get_region_by_id(1);
+    region.wait().unwrap();
```
Comment (BusyJay) [inline tests/pd/test_rpc_client.rs]: How to verify that the client is reconnected to remote again?

Comment (overvenus) [inline tests/pd/test_rpc_client.rs]: If they are not connected, unwrap fails.

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**